gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -213,7 +213,7 @@ private ManagedChannel createSingleChannel() throws IOException {
     GrpcMetadataHandlerInterceptor metadataHandlerInterceptor =
         new GrpcMetadataHandlerInterceptor();
 
-    int colon = endpoint.indexOf(':');
+    int colon = endpoint.lastIndexOf(':');
```

Comment:
It looks like in case if there is no port at all, this code will try to treat the last segment of ipv6 as port number. For example in
`2001:db8:85a3:8d3:1319:8a2e:370:7348`
7348 becomes a port number.

I understand that it is unlikely case, but please consider handling it (i.e. add proper logic for both `ipv4` and `ipv6` cases, also `colon` is porbably a bad name here ans it should clearly indicate that we are searching for a port here).

Reply:
The usage of `port` below ([1](https://github.com/googleapis/gax-java/pull/768/files#diff-25ba434a47d99bcc0b998965a298661cR228) and [2](https://github.com/googleapis/gax-java/pull/768/files#diff-25ba434a47d99bcc0b998965a298661cR234)) explicitly necessitates a port being provided as part of `endpoint` since there is no form of default port handling due to not having scheme be part of `endpoint`. All tests defined in the corresponding test class always specify `hostname:port`.  

I think if we wanted to update this class to support connections without port specified it'd be a lot more work than this.

Reply:
In fact, there is an existing test [`testEndpointNoPort`](https://github.com/googleapis/gax-java/blob/81b3cbeaeb1d42e360f3f344fb27b9cef28b282a/gax-grpc/src/test/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProviderTest.java#L75) that expects an `IllegalArgumentException` in the case no port is specified.